kafka-client/src/main/java/dev/responsive/kafka/api/stores/ResponsiveKeyValueParams.java

```diff
@@ -46,7 +46,16 @@ public static ResponsiveKeyValueParams fact(final String name) {
   }
 
   public ResponsiveKeyValueParams withTimeToLive(final Duration timeToLive) {
-    this.timeToLive = timeToLive;
+    return withTtlProvider(TtlProvider.withDefault(timeToLive));
+  }
+
+  public ResponsiveKeyValueParams withTtlProvider(final TtlProvider<?, ?> ttlProvider) {
+    if (ttlProvider.hasConstantTtl() && !ttlProvider.defaultTtl().isFinite()) {
+      throw new IllegalArgumentException("Passed in ttlProvider with infinite ttl, "
+                                             + "this is equivalent to no ttl and you "
+                                             + "should not set the ttlProvider for this");
```

Comment:
just wondering, why not just do nothing (i.e. leave it as Optional.empty and just return early)? I'd rather not throw a bunch of errors as that makes it stricter in the way we allow usage (I actually like having "NOOP" instances instead of nulls/empty in Java, so as a user, especially in tests, I may consider doing that)

Reply:
oh I actually would prefer to just silently convert it! I had the impression from previous interactions that you strongly preferred to throw an exception when the user does something that doesn't make sense rather than just correct it? Anyways I'm happy to change it

Reply:
Now that you mention it... 😆 In cases where it's ambiguous I'd prefer throwing an error, but in this case I think we _are_ doing what they want, we're just optimizing it away underneath the hood. (i.e. it's not invalid to set a default + infinite TTL, it's just the same as the default behavior)